pkg/protocol/http1/client.go

```diff
@@ -508,6 +508,17 @@ func (c *HostClient) doNonNilReqResp(req *protocol.Request, resp *protocol.Respo
 	}
 	// error happened when writing request, close the connection, and try another connection if retry is enabled
 	if err != nil {
+		// Try to read the response body if the server has sent it.
+		zr := c.acquireReader(conn)
+		// If the server has sent the response body, parse it and return.
+		// In this case, the request is considered as successful.
+		// Otherwise, return the former error.
+		if respI.ReadHeaderAndLimitBody(resp, zr, c.MaxResponseBodySize) == nil {
+			zr.Release()
+			c.closeConn(cc)
+			return false, nil
```

Comment:
I don't think it's an appropriate way to handle responses because it's actually an error. For exist users, they handle this case by judging err previously. However, they need to parse the unexpected body data. It's a break change.